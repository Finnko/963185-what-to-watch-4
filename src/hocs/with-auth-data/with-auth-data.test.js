import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Input from '../../components/input/input.tsx';
import withAuthData from './with-auth-data';

const inputNames = {
  email: `Email address`,
  password: `Password`,
};

const MockComponent = ({
  formControls,
  isFormValid,
  onInputChange,
}) => {
  const renderInputs = () => {
    return Object.keys(formControls).map((name, index) => {
      const {
        value,
        valid,
        touched,
        validation
      } = formControls[name];

      return (
        <Input
          key={name + index}
          id={`user-${name}`}
          name={name}
          label={inputNames[name]}
          placeholder={inputNames[name]}
          value={value}
          valid={valid}
          touched={touched}
          shouldValidate={!!validation}
          onInputChange={onInputChange}
        />
      );
    });
  };

  return (
    <div>
      <div className="sign-in__fields">
        {renderInputs()}
      </div>

      <div className="sign-in__submit">
        <button
          className="sign-in__btn"
          type="submit"
          disabled={!isFormValid}
        >
        </button>

      </div>
    </div>
  );
};

MockComponent.propTypes = {
  formControls: PropTypes.object,
  isFormValid: PropTypes.bool,
  onInputChange: PropTypes.func,
};

const MockComponentWrapped = withAuthData(MockComponent);

describe(`Should hoc withAuthData render correctly`, () => {
  it(`Snapshot render withAuthData`, () => {
    const tree = renderer.create(<MockComponentWrapped />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
