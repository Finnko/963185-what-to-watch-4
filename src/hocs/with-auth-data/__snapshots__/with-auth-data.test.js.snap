// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should hoc withAuthData render correctly Snapshot render withAuthData 1`] = `
<div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field "
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      disabled={true}
      type="submit"
    />
  </div>
</div>
`;
